### ISSUE TITLE
Support cohere / jina compatible /v1/rerank in server?

### ISSUE THREAD

**Issue by theaerotoad** [CrispStrobe/CrispEmbed#51]
Right now, `POST /rerank` uses a (logical) custom schema (returning raw `score` logits and flat string docs). Wondering if it'd be good to add a  `/v1/rerank` endpoint that follows the de facto Cohere / Jina rerank standard?

You could have `/v1/rerank` live alongside the existing `/rerank`.  That'd make `crispembed-server` a drop-in replacement for tools like LiteLLM, LlamaIndex, and LangChain that expect the standard schema.  Think it'd also work well with `llama-swap`, which respects /v1/rerank.

I think the main differences from the standard format are (pulling from AI here):

* **Endpoint path:** `POST /v1/rerank`
* **Score key & normalization:** `relevance_score` (typically sigmoid-scaled between 0.0 and 1.0) instead of raw `score` logits.
* **Document format:** Expects `document: {"text": "..."}` in the results object (often controlled via a `"return_documents": true/false` request flag).
* **Request payload:** Accepting `"model": "..."` (even if ignored/optional), and accepting `documents` as either a list of strings or a list of `{"text": "..."}` objects.

IIRC, the response format looks like:

```json
{
  "results": [
    {
      "index": 1,
      "relevance_score": 0.9998,
      "document": {
        "text": "Paris is the capital and most populous city of France."
      }
    }
  ]
}

```

Think the other main change is the need to flatten the rankings with a sigmoid.  

I'm happy to try to code this up, too and provide a patch.  Either way, thanks for sharing these awesome swiss-army-knife tools!

**Comment by CrispStrobe** [comment]
Good suggestion — implemented on `main`, so no need to write the patch. Thanks for offering, and for spelling out the schema differences; that list was accurate and is essentially what shipped.

`POST /v1/rerank` now lives alongside `/rerank`, which keeps its existing contract unchanged (raw `score`, flat `document` string).

```bash
curl http://localhost:8080/v1/rerank -H 'Content-Type: application/json' -d '{
  "model": "bge-reranker-v2-m3",
  "query": "What is the capital of France?",
  "documents": ["Bananas are a good source of potassium.",
                "Paris is the capital and most populous city of France."],
  "top_n": 2, "return_documents": true }'
```

Against `ettin-reranker-150m-v1` (the model from your other issue):

```json
{"id": "rerank-0", "model": "ettin-reranker-150m-v1-q4_k", "object": "list",
 "results": [{"index": 1, "relevance_score": 0.999981991},
             {"index": 0, "relevance_score": 0.005674335}],
 "meta": {"api_version": {"version": "2"}, "billed_units": {"search_units": 1}}}
```

On the four points you raised:

- **`documents`** accepts both `["text", …]` and `[{"text": "…"}, …]`. Worth noting since you offered to patch it: the obvious implementation is wrong here. The existing string extractor flattens *every* string inside the array, so the object form parses as `["text","a","text","b"]` — twice as many documents, all garbage, with indices that mean nothing to the caller. It needed an element-aware walk; there's a unit test that asserts the flat parser really does return 4 for a 2-document payload, so nobody re-introduces it.
- **`relevance_score`** is `sigmoid(logit)`, as you expected. It's emitted at 9 decimal places rather than the 6 the native route uses: a confident cross-encoder logit near +15 saturates sigmoid past 0.9999995, and at 6 dp several top hits all print `1.000000`, so a client that re-sorts by score would lose the ordering the endpoint just computed. `CRISPEMBED_SERVER_RERANK_RAW_SCORES=1` emits the raw logit instead, if you want to compare the two routes number-for-number.
- **`return_documents`** defaults to `false`, matching Cohere. Set it `true` to get `document: {"text": …}`.
- **`model`** is accepted and echoed back, and ignored for routing — the server serves whatever GGUF it was started with.

One deliberate omission: there's no `usage` / token-count field. The pair-encoder path doesn't expose a token count, and emitting a plausible-looking number seemed worse than leaving an optional field out. If some client you're using requires it, say so and I'll look at wiring a real count rather than a fake one.

Ordering is identical to `/rerank` — sigmoid is monotonic, so it can't reorder anything, and there's a live test that asserts the two routes agree on the ranking rather than just trusting the maths. Please do report back if llama-swap or LiteLLM trips on anything.

**Comment by theaerotoad** [comment]
Fantastic! Will check it out today in my chain with llamaswap. Thank you.
